pandas/io/pytables.py

```diff
@@ -2968,11 +2968,17 @@ def write_array_empty(self, key: str, value: ArrayLike):
         node._v_attrs.value_type = str(value.dtype)
         node._v_attrs.shape = value.shape
 
-    def write_array(self, key: str, value: ArrayLike, items: Optional[Index] = None):
-        # TODO: we only have one test that gets here, the only EA
+    def write_array(self, key: str, obj: Series, items: Optional[Index] = None):
+        # TODO: we only have a few tests that get here, the only EA
         #  that gets passed is DatetimeArray, and we never have
         #  both self._filters and EA
-        assert isinstance(value, (np.ndarray, ABCExtensionArray)), type(value)
+
+        if isinstance(obj, (np.ndarray, ABCExtensionArray)):
```

Comment:
is the annotation of obj as a Series accurate?  if so, this branch is never hit

Reply:
No it's not. Should be SeriesOrFrame (will fix)